spring-cloud-kubernetes-discovery/src/main/java/org/springframework/cloud/kubernetes/discovery/KubernetesCatalogWatchAutoConfiguration.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2013-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.kubernetes.discovery;
+
+import io.fabric8.kubernetes.client.KubernetesClient;
+
+import org.springframework.boot.autoconfigure.AutoConfigureAfter;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.cloud.client.ConditionalOnDiscoveryEnabled;
+import org.springframework.cloud.kubernetes.KubernetesAutoConfiguration;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.scheduling.annotation.EnableScheduling;
+
+/**
+ * Auto configuration for catalog watcher.
+ *
+ * @author Tim Ysewyn
+ */
+@Configuration
+@EnableScheduling
```

Comment:
We decided not to do this.

Reply:
Do you remember why? This is the only service registry implementation that needs that annotation explicitly.

Reply:
We chose to document it instead.

Reply:
See https://github.com/spring-cloud/spring-cloud-kubernetes/issues/385

Reply:
Thanks for the input! So what if we do it like it's done in the [ConsulCatalogWatchAutoConfiguration](https://github.com/spring-cloud/spring-cloud-consul/blob/master/spring-cloud-consul-discovery/src/main/java/org/springframework/cloud/consul/discovery/ConsulCatalogWatchAutoConfiguration.java)?
Then we don't need to explicitly add `@EnableScheduling`, and it will only be created when the watcher is enabled with `spring.cloud.kubernetes.discovery.catalog-services-watch.enabled`.

Maybe not for this PR, but could be a nice improvement for the next minor release.

Reply:
Yup, not for this PR.